File: dailyChallenges/0019-ValidAnagram/19-ValidAnagram.go
func isAnagram(s string, t string) bool {
	sMap := make(map[rune]int)
	tMap := make(map[rune]int)

	if len(s) != len(t) {
		return false
	}

	for _, char := range s {
		sMap[char]++
	}

	for _, char := range t {
		tMap[char]++
	}

	for key, value1 := range sMap {
		if value2, ok := tMap[key]; !ok || value1 != value2 {
			return false
		}
	}
	return true
}

File: dailyChallenges/0019-ValidAnagram/19-ValidAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> smap;
        unordered_map<char,int> tmap;

        if(s.size() != t.size())
        {
            return false;
        }

        for(int i=0; i < s.size(); i++)
        {
            smap[s[i]]++;
            tmap[t[i]]++;
        }

        for(int i=0; i < smap.size(); i++)
        {
            if(smap[i] != tmap[i])
            {
                return false;
            }
        }
        return true;
    }
};